round trip error

@rly when I try to run the round trip test here I get the following error:

```pytb
src/pynwb/tests/test_anatomical_coordinates.py:8 (test_create_anatomical_coordinates_table)
self = <pynwb.io.core.NWBContainerMapper object at 0x146f9e1b0>, kwargs = {}
builder = root/general/localization/space GroupBuilder {'attributes': {'namespace': 'ndx-anatomical-localization', 'neurodata_ty...e middle of the anterior commissure', 'space_name': 'CCFv3', 'units': 'um'}, 'groups': {}, 'datasets': {}, 'links': {}}
manager = <hdmf.build.manager.BuildManager object at 0x147f6f230>
parent = {'source': '/Users/bendichter/dev/ndx-anatomical-localization/src/pynwb/tests/test.nwb', 'location': 'root/localization', 'namespace': 'ndx-anatomical-localization', 'data_type': 'Localization'}
cls = <class 'ndx_anatomical_localization.ndx_anatomical_localization.Space'>
subspecs = {{'doc': 'name of the space', 'name': 'space_name', 'dtype': 'text'}: 'CCFv3', {'doc': 'origin of the space', 'name': ...ior (a.k.a dorsal). For dorsal/ventral use 'S/I' (superior/inferior).", 'name': 'orientation', 'dtype': 'text'}: 'RAS'}
const_args = {'orientation': 'RAS', 'origin': 'In the middle of the anterior commissure', 'space_name': 'CCFv3', 'units': 'um'}
subspec = {'doc': "A 3-letter string. One of APLRSI for each of x, y, and z. For example, the most common orientation is 'RAS', ...is superior (a.k.a dorsal). For dorsal/ventral use 'S/I' (superior/inferior).", 'name': 'orientation', 'dtype': 'text'}

    @docval({'name': 'builder', 'type': (DatasetBuilder, GroupBuilder),
             'doc': 'the builder to construct the AbstractContainer from'},
            {'name': 'manager', 'type': BuildManager, 'doc': 'the BuildManager for this build'},
            {'name': 'parent', 'type': (Proxy, AbstractContainer),
             'doc': 'the parent AbstractContainer/Proxy for the AbstractContainer being built', 'default': None})
    def construct(self, **kwargs):
        ''' Construct an AbstractContainer from the given Builder '''
        builder, manager, parent = getargs('builder', 'manager', 'parent', kwargs)
        cls = manager.get_cls(builder)
        # gather all subspecs
        subspecs = self.__get_subspec_values(builder, self.spec, manager)
        # get the constructor argument that each specification corresponds to
        const_args = dict()
        # For Data container classes, we need to populate the data constructor argument since
        # there is no sub-specification that maps to that argument under the default logic
        if issubclass(cls, Data):
            if not isinstance(builder, DatasetBuilder):  # pragma: no cover
                raise ValueError('Can only construct a Data object from a DatasetBuilder - got %s' % type(builder))
            const_args['data'] = self.__check_ref_resolver(builder.data)
        for subspec, value in subspecs.items():
            const_arg = self.get_const_arg(subspec)
            if const_arg is not None:
                if isinstance(subspec, BaseStorageSpec) and subspec.is_many():
                    existing_value = const_args.get(const_arg)
                    if isinstance(existing_value, list):
                        value = existing_value + value
                const_args[const_arg] = value
        # build kwargs for the constructor
        kwargs = dict()
        for const_arg in get_docval(cls.__init__):
            argname = const_arg['name']
            override = self.__get_override_carg(argname, builder, manager)
            if override is not None:
                val = override
            elif argname in const_args:
                val = const_args[argname]
            else:
                continue
            kwargs[argname] = val
        try:
>           obj = self.__new_container__(cls, builder.source, parent, builder.attributes.get(self.__spec.id_key()),
                                         **kwargs)

../../../../hdmf/src/hdmf/build/objectmapper.py:1262: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pynwb.io.core.NWBContainerMapper object at 0x146f9e1b0>
cls = <class 'ndx_anatomical_localization.ndx_anatomical_localization.Space'>
container_source = '/Users/bendichter/dev/ndx-anatomical-localization/src/pynwb/tests/test.nwb'
parent = {'source': '/Users/bendichter/dev/ndx-anatomical-localization/src/pynwb/tests/test.nwb', 'location': 'root/localization', 'namespace': 'ndx-anatomical-localization', 'data_type': 'Localization'}
object_id = 'e3029917-a579-4e93-a6b0-96a06fd80867', kwargs = {}
obj = <[AttributeError("'Space' object has no attribute '_AbstractContainer__name'") raised in repr()] Space object at 0x1683d5880>

    def __new_container__(self, cls, container_source, parent, object_id, **kwargs):
        """A wrapper function for ensuring a container gets everything set appropriately"""
        obj = cls.__new__(cls, container_source=container_source, parent=parent, object_id=object_id,
                          in_construct_mode=True)
        # obj has been created and is in construction mode, indicating that the object is being constructed by
        # the automatic construct process during read, rather than by the user
>       obj.__init__(**kwargs)
E       TypeError: Space.__init__() missing 5 required positional arguments: 'name', 'space_name', 'origin', 'units', and 'orientation'

../../../../hdmf/src/hdmf/build/objectmapper.py:1275: TypeError

The above exception was the direct cause of the following exception:

    def test_create_anatomical_coordinates_table():
    
        nwbfile = mock_NWBFile()
    
        localization = Localization()
        nwbfile.add_lab_meta_data([localization])
    
        electrodes_table = mock_ElectrodeTable(nwbfile=nwbfile)
    
        space = Space.get_predefined_space("CCFv3")
        localization.add_spaces([space])
    
        table = AnatomicalCoordinatesTable(
            name="MyAnatomicalLocalization",
            target=electrodes_table,
            description="Anatomical coordinates table",
            method="method",
            space=space,
        )
        [table.add_row(x=1.0, y=2.0, z=3.0, brain_region="CA1", target_object=x) for x in range(5)]
    
        localization.add_anatomical_coordinates_tables([table])
    
        with NWBHDF5IO("test.nwb", "w") as io:
            io.write(nwbfile)
    
        with NWBHDF5IO("test.nwb", "r", load_namespaces=True) as io:
>           read_nwbfile = io.read()

test_anatomical_coordinates.py:36: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../../hdmf/src/hdmf/utils.py:668: in func_call
    return func(args[0], **pargs)
/opt/miniconda3/lib/python3.12/site-packages/pynwb/__init__.py:352: in read
    file = super().read(**kwargs)
../../../../hdmf/src/hdmf/backends/hdf5/h5tools.py:500: in read
    return super().read(**kwargs)
../../../../hdmf/src/hdmf/utils.py:668: in func_call
    return func(args[0], **pargs)
../../../../hdmf/src/hdmf/backends/io.py:60: in read
    container = self.__manager.construct(f_builder)
../../../../hdmf/src/hdmf/utils.py:668: in func_call
    return func(args[0], **pargs)
../../../../hdmf/src/hdmf/build/manager.py:286: in construct
    result = self.__type_map.construct(builder, self, None)
../../../../hdmf/src/hdmf/utils.py:668: in func_call
    return func(args[0], **pargs)
../../../../hdmf/src/hdmf/build/manager.py:814: in construct
    return obj_mapper.construct(builder, build_manager, parent)
../../../../hdmf/src/hdmf/utils.py:668: in func_call
    return func(args[0], **pargs)
../../../../hdmf/src/hdmf/build/objectmapper.py:1232: in construct
    subspecs = self.__get_subspec_values(builder, self.spec, manager)
../../../../hdmf/src/hdmf/build/objectmapper.py:1161: in __get_subspec_values
    self.__get_sub_builders(groups, spec.groups, manager, ret)
../../../../hdmf/src/hdmf/build/objectmapper.py:1212: in __get_sub_builders
    ret.update(self.__get_subspec_values(sub_builder, subspec, manager))
../../../../hdmf/src/hdmf/build/objectmapper.py:1161: in __get_subspec_values
    self.__get_sub_builders(groups, spec.groups, manager, ret)
../../../../hdmf/src/hdmf/build/objectmapper.py:1204: in __get_sub_builders
    sub_builder = self.__flatten(sub_builder, subspec, manager)
../../../../hdmf/src/hdmf/build/objectmapper.py:1217: in __flatten
    tmp = [manager.construct(b) for b in sub_builder]
../../../../hdmf/src/hdmf/utils.py:668: in func_call
    return func(args[0], **pargs)
../../../../hdmf/src/hdmf/build/manager.py:282: in construct
    result = self.__type_map.construct(builder, self, parent)
../../../../hdmf/src/hdmf/utils.py:668: in func_call
    return func(args[0], **pargs)
../../../../hdmf/src/hdmf/build/manager.py:814: in construct
    return obj_mapper.construct(builder, build_manager, parent)
../../../../hdmf/src/hdmf/utils.py:668: in func_call
    return func(args[0], **pargs)
../../../../hdmf/src/hdmf/build/objectmapper.py:1232: in construct
    subspecs = self.__get_subspec_values(builder, self.spec, manager)
../../../../hdmf/src/hdmf/build/objectmapper.py:1161: in __get_subspec_values
    self.__get_sub_builders(groups, spec.groups, manager, ret)
../../../../hdmf/src/hdmf/build/objectmapper.py:1204: in __get_sub_builders
    sub_builder = self.__flatten(sub_builder, subspec, manager)
../../../../hdmf/src/hdmf/build/objectmapper.py:1217: in __flatten
    tmp = [manager.construct(b) for b in sub_builder]
../../../../hdmf/src/hdmf/utils.py:668: in func_call
    return func(args[0], **pargs)
../../../../hdmf/src/hdmf/build/manager.py:282: in construct
    result = self.__type_map.construct(builder, self, parent)
../../../../hdmf/src/hdmf/utils.py:668: in func_call
    return func(args[0], **pargs)
../../../../hdmf/src/hdmf/build/manager.py:814: in construct
    return obj_mapper.construct(builder, build_manager, parent)
../../../../hdmf/src/hdmf/utils.py:668: in func_call
    return func(args[0], **pargs)
../../../../hdmf/src/hdmf/build/objectmapper.py:1232: in construct
    subspecs = self.__get_subspec_values(builder, self.spec, manager)
../../../../hdmf/src/hdmf/build/objectmapper.py:1155: in __get_subspec_values
    ret[subspec] = manager.construct(links[subspec.name].builder)
../../../../hdmf/src/hdmf/utils.py:668: in func_call
    return func(args[0], **pargs)
../../../../hdmf/src/hdmf/build/manager.py:282: in construct
    result = self.__type_map.construct(builder, self, parent)
../../../../hdmf/src/hdmf/utils.py:668: in func_call
    return func(args[0], **pargs)
../../../../hdmf/src/hdmf/build/manager.py:814: in construct
    return obj_mapper.construct(builder, build_manager, parent)
../../../../hdmf/src/hdmf/utils.py:668: in func_call
    return func(args[0], **pargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pynwb.io.core.NWBContainerMapper object at 0x146f9e1b0>, kwargs = {}
builder = root/general/localization/space GroupBuilder {'attributes': {'namespace': 'ndx-anatomical-localization', 'neurodata_ty...e middle of the anterior commissure', 'space_name': 'CCFv3', 'units': 'um'}, 'groups': {}, 'datasets': {}, 'links': {}}
manager = <hdmf.build.manager.BuildManager object at 0x147f6f230>
parent = {'source': '/Users/bendichter/dev/ndx-anatomical-localization/src/pynwb/tests/test.nwb', 'location': 'root/localization', 'namespace': 'ndx-anatomical-localization', 'data_type': 'Localization'}
cls = <class 'ndx_anatomical_localization.ndx_anatomical_localization.Space'>
subspecs = {{'doc': 'name of the space', 'name': 'space_name', 'dtype': 'text'}: 'CCFv3', {'doc': 'origin of the space', 'name': ...ior (a.k.a dorsal). For dorsal/ventral use 'S/I' (superior/inferior).", 'name': 'orientation', 'dtype': 'text'}: 'RAS'}
const_args = {'orientation': 'RAS', 'origin': 'In the middle of the anterior commissure', 'space_name': 'CCFv3', 'units': 'um'}
subspec = {'doc': "A 3-letter string. One of APLRSI for each of x, y, and z. For example, the most common orientation is 'RAS', ...is superior (a.k.a dorsal). For dorsal/ventral use 'S/I' (superior/inferior).", 'name': 'orientation', 'dtype': 'text'}

    @docval({'name': 'builder', 'type': (DatasetBuilder, GroupBuilder),
             'doc': 'the builder to construct the AbstractContainer from'},
            {'name': 'manager', 'type': BuildManager, 'doc': 'the BuildManager for this build'},
            {'name': 'parent', 'type': (Proxy, AbstractContainer),
             'doc': 'the parent AbstractContainer/Proxy for the AbstractContainer being built', 'default': None})
    def construct(self, **kwargs):
        ''' Construct an AbstractContainer from the given Builder '''
        builder, manager, parent = getargs('builder', 'manager', 'parent', kwargs)
        cls = manager.get_cls(builder)
        # gather all subspecs
        subspecs = self.__get_subspec_values(builder, self.spec, manager)
        # get the constructor argument that each specification corresponds to
        const_args = dict()
        # For Data container classes, we need to populate the data constructor argument since
        # there is no sub-specification that maps to that argument under the default logic
        if issubclass(cls, Data):
            if not isinstance(builder, DatasetBuilder):  # pragma: no cover
                raise ValueError('Can only construct a Data object from a DatasetBuilder - got %s' % type(builder))
            const_args['data'] = self.__check_ref_resolver(builder.data)
        for subspec, value in subspecs.items():
            const_arg = self.get_const_arg(subspec)
            if const_arg is not None:
                if isinstance(subspec, BaseStorageSpec) and subspec.is_many():
                    existing_value = const_args.get(const_arg)
                    if isinstance(existing_value, list):
                        value = existing_value + value
                const_args[const_arg] = value
        # build kwargs for the constructor
        kwargs = dict()
        for const_arg in get_docval(cls.__init__):
            argname = const_arg['name']
            override = self.__get_override_carg(argname, builder, manager)
            if override is not None:
                val = override
            elif argname in const_args:
                val = const_args[argname]
            else:
                continue
            kwargs[argname] = val
        try:
            obj = self.__new_container__(cls, builder.source, parent, builder.attributes.get(self.__spec.id_key()),
                                         **kwargs)
        except Exception as ex:
            msg = 'Could not construct %s object due to: %s' % (cls.__name__, ex)
>           raise ConstructError(builder, msg) from ex
E           hdmf.build.errors.ConstructError: (root/general/localization/space GroupBuilder {'attributes': {'namespace': 'ndx-anatomical-localization', 'neurodata_type': 'Space', 'object_id': 'e3029917-a579-4e93-a6b0-96a06fd80867', 'orientation': 'RAS', 'origin': 'In the middle of the anterior commissure', 'space_name': 'CCFv3', 'units': 'um'}, 'groups': {}, 'datasets': {}, 'links': {}}, "Could not construct Space object due to: Space.__init__() missing 5 required positional arguments: 'name', 'space_name', 'origin', 'units', and 'orientation'")

../../../../hdmf/src/hdmf/build/objectmapper.py:1266: ConstructError
```

do you know what might be causing this?

--- a/spec/ndx-anatomical-localization.extensions.yaml
+++ b/spec/ndx-anatomical-localization.extensions.yaml
@@ -27,12 +27,15 @@ groups:
         dtype: text
         required: true
       - name: orientation
-        doc: "A 3-letter string. One of APLRSI for each of x, y, and z. For example, the most common orientation is 'RAS', which means x is right, y is anterior, and z is superior (a.k.a dorsal). For dorsal/ventral use 'S/I' (superior/inferior)."
+        doc: "A 3-letter string. One of APLRSI for each of x, y, and z. For example, the most common
+          orientation is 'RAS', which means x is right, y is anterior, and z is superior (a.k.a. dorsal).
+          For dorsal/ventral use 'S/I' (superior/inferior). In the AnatomicalCoordinatesTable, an orientation of
+          'RAS' corresponds to coordinates in the order of (ML (x), AP (y), DV (z))."
         dtype: text
         required: true
   - neurodata_type_def: AnatomicalCoordinatesTable
     neurodata_type_inc: DynamicTable
-    doc: "A table for storing coordinates"
+    doc: "A table for storing anatomical coordinates for rows of another table determined using a single method"
     links:
       - target_type: Space
         quantity: 1

--- a/src/pynwb/ndx_anatomical_localization/__init__.py
+++ b/src/pynwb/ndx_anatomical_localization/__init__.py
@@ -8,14 +8,12 @@ except ImportError:
     from importlib_resources import files
 
 # # Get path to the namespace.yaml file with the expected location when installed not in editable mode
-# __location_of_this_file = files(__name__)
-# __spec_path = __location_of_this_file / "spec" / "ndx-anatomical-localization.namespace.yaml"
+__location_of_this_file = files(__name__)
+__spec_path = __location_of_this_file / "spec" / "ndx-anatomical-localization.namespace.yaml"
 
 # # If that path does not exist, we are likely running in editable mode. Use the local path instead
-# if not os.path.exists(__spec_path):
-#     __spec_path = __location_of_this_file.parent.parent.parent / "spec" / "ndx-anatomical-localization.namespace.yaml"
-
-__spec_path = "/Users/bendichter/dev/ndx-anatomical-localization/spec/ndx-anatomical-localization.namespace.yaml"
+if not os.path.exists(__spec_path):
+    __spec_path = __location_of_this_file.parent.parent.parent / "spec" / "ndx-anatomical-localization.namespace.yaml"
 
 # Load the namespace
 load_namespaces(str(__spec_path))

--- a/src/pynwb/ndx_anatomical_localization/ndx_anatomical_localization.py
+++ b/src/pynwb/ndx_anatomical_localization/ndx_anatomical_localization.py
@@ -1,5 +1,6 @@
 from hdmf.common import DynamicTable
-from pynwb import get_class, register_class, docval
+from hdmf.utils import docval, get_docval, AllowPositional
+from pynwb import get_class, register_class
 
 
 TempSpace = get_class("Space", "ndx-anatomical-localization")
@@ -26,6 +27,7 @@ class Space(TempSpace):
         {"name": "origin", "type": str, "doc": "origin of the space"},
         {"name": "units", "type": str, "doc": "units of the space"},
         {"name": "orientation", "type": str, "doc": "orientation of the space"},
+        allow_positional=AllowPositional.ERROR,
     )
     def __init__(self, name, space_name, origin, units, orientation):
         assert len(orientation) == 3, "orientation must be a string of length 3"
@@ -51,13 +53,23 @@ class Space(TempSpace):
 class AnatomicalCoordinatesTable(TempAnatomicalCoordinatesTable):
 
     @docval(
-        {"name": "name", "type": str, "doc": "name of the table"},
-        {"name": "target", "type": DynamicTable, "doc": "target table"},
-        {"name": "description", "type": str, "doc": "description of the table"},
         {"name": "space", "type": Space, "doc": "space of the table"},
         {"name": "method", "type": str, "doc": "method of the table"},
+        {"name": "target", "type": DynamicTable,
+         "doc": 'target table. ignored if a "target_object" column is provided in "columns"', "default": None},
+        *get_docval(DynamicTable.__init__),
+        allow_positional=AllowPositional.ERROR,
     )
     def __init__(self, **kwargs):
+        columns = kwargs.get("columns")
         target = kwargs.pop("target")
+        if not columns or "target_object" not in [c.name for c in columns]:
+            # set the target table of the "target_object" column only if the "target_object" column is not in "columns"
+            if target is None:
+                raise ValueError(
+                    '"target" (the target table that contains the objects that have these coordinates) '
+                    'must be provided in AnatomicalCoordinatesTable.__init__ '
+                    'if the "target_object" column is not in "columns".')
+            kwargs["target_tables"] = {"target_object": target}
+
         super().__init__(**kwargs)
-        self.target_object.table = target

--- a/src/pynwb/tests/test_anatomical_coordinates.py
+++ b/src/pynwb/tests/test_anatomical_coordinates.py
@@ -2,6 +2,8 @@
 from pynwb import NWBHDF5IO
 from pynwb.testing.mock.file import mock_NWBFile
 from pynwb.testing.mock.ecephys import mock_ElectrodeTable
+import numpy as np
+import numpy.testing as npt
 
 from ndx_anatomical_localization import AnatomicalCoordinatesTable, Space, Localization
 
@@ -34,5 +36,17 @@ def test_create_anatomical_coordinates_table():
 
     with NWBHDF5IO("test.nwb", "r", load_namespaces=True) as io:
         read_nwbfile = io.read()
+        read_electrodes_table = read_nwbfile.electrodes
+        read_localization = read_nwbfile.lab_meta_data["localization"]
 
-        assert read_nwbfile.lab_meta_data["localization"]["MyAnatomicalLocalization"].space.space_name == "CCFv3"
+        read_coordinates_table = read_localization.anatomical_coordinates_tables["MyAnatomicalLocalization"]
+
+        assert read_coordinates_table.method == "method"
+        assert read_coordinates_table.description == "Anatomical coordinates table"
+        assert read_coordinates_table.target_object.table is read_electrodes_table
+        assert read_coordinates_table.space.fields == Space.get_predefined_space("CCFv3").fields
+        npt.assert_array_equal(read_coordinates_table["x"].data[:], np.array([1.0, 1.0, 1.0, 1.0, 1.0]))
+        npt.assert_array_equal(read_coordinates_table["y"].data[:], np.array([2.0, 2.0, 2.0, 2.0, 2.0]))
+        npt.assert_array_equal(read_coordinates_table["z"].data[:], np.array([3.0, 3.0, 3.0, 3.0, 3.0]))
+        npt.assert_array_equal(read_coordinates_table["brain_region"].data[:], np.array(["CA1", "CA1", "CA1", "CA1", "CA1"]))
+        npt.assert_array_equal(read_coordinates_table["target_object"].data[:], np.array([0, 1, 2, 3, 4]))

